### PR TITLE
Housing backend

### PR DESCRIPTION
**Pull request by annarog**
Ref #13
Added schemas, routes, and tests for housing resources persistence.

### CODE REVIEW

File: express-server/models/Housing.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+const mongoose = require('mongoose');
+const Schema = mongoose.Schema;
+const Resource = require('../models/Resource');
+
+const housingSchema = new Schema({
+    term: { type: String, required: true},
+    gender: String,
+    constraints: String
```
Comment (Ldhatter) [inline express-server/models/Housing.js]: would constaints maybe be an array of strings?
eg: [ "No children", "No alcohol" ...] ?

Comment (Ldhatter) [reply express-server/models/Housing.js]: also how does this schema inherit/make use of the resourceSchema?

Comment (Ldhatter) [reply express-server/models/Housing.js]: ok sweet @annarog thanks for the clarification on the discriminator!

Comment (Ldhatter) [reply express-server/models/Housing.js]: as for constraints. could set it to an array of strings (that way the frontend can make use of list of chips for example). but we could do some research + ask James what they normally make note of for housing and refactor after.

File: express-server/server.js [hunk 2]
```diff
@@ -33,6 +34,7 @@ app.use(cors({
 app.use('/', api);
 app.use('/user', user);
 app.use('/participant', participant);
+app.use('/resource', resource);
 app.use(cors())
 
 //Connect to mongo
```
Comment (Ldhatter) [inline express-server/server.js]: Can we change the credentials and add to env variable.

Comment (annarog) [reply express-server/server.js]: It's already done in another branch. We should definitely change the credentials once everything's merged though

**Comment by Ldhatter** [comment]
@annarog just wondering can you explain a bit how the travis.yml works with the mongo credentials?

**Comment by annarog** [comment]
@Ldhatter I ran `travis encrypt MONGOLAB_URL=our_mongo_credentials_url --add env` which encrypts the URL and adds the encrypted version to the yml file (it's the `env: secure: ...` part). 
Then whenever travis runs, it decrypts this value to the actual mongo url :)

**Pull request merged**